code/game/area/areas.dm

```diff
@@ -403,7 +403,8 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 /area/Entered(atom/movable/arrived, area/old_area)
 	set waitfor = FALSE
 	SEND_SIGNAL(src, COMSIG_AREA_ENTERED, arrived, old_area)
-	if(!LAZYACCESS(arrived.important_recursive_contents, RECURSIVE_CONTENTS_AREA_SENSITIVE))
+
+	if(!arrived.important_recursive_contents?[RECURSIVE_CONTENTS_AREA_SENSITIVE])
```

Comment:
why?

Reply:
lazyaccess is dumb and is designed for safety in a way thats arguably good in some contexts and completely useless in others while slowing down runtime in all contexts. since we're not checking a numerical index part of lazyaccess is redundant

Reply:
is this code hot enough that avoiding the isnum is worth it? how large is the difference

Reply:
![Screenshot_2085](https://user-images.githubusercontent.com/15794172/175843657-ea80d09b-1bfd-4681-8e33-3410d1831ebc.png)
about 20% faster for this case not including the offsets from both being datum vars or ~90 nanoseconds per iteration. in that time, a photon of light could travel a whole 90 feet in a vacuum. a whole 90 feet! over the few tens of thousands of times this proc is called over the course of a round that photon could travel a few hundred miles. what amazing savings!

in seriousness i still dont like lazyaccess because it provides no actual value when you know youre reading an associative value and very debatable value when youre reading a value at a specific numerical index (shouldnt you know whether the list is that big, and if it isnt shouldnt you get an error?). 

the lazy macros should just be list getters and setters that are safe regardless of whether the list is null or not, and specifically for setters also null the list if totally empty